lib/pcp/client.rb

```diff
@@ -53,6 +53,11 @@ def connect(seconds = 0)
         raise "Cannot be run on the same thread as the reactor"
       end
 
+      if @connection
+        # We close over so much, we really just need to say no for now
+        raise "Can only connect once per client"
```

Comment:
If this might show up in the middle of a large beaker run with several components in play; it might be helpful to be more specific and say "A ruby-pcp-client instance can only connect once".
Though I guess this error is likely to only appear during the development of new code and the code author should find it self-explanatory